pandas/core/indexes/range.py

```diff
@@ -1102,3 +1103,49 @@ def _arith_method(self, other, op):
         except (ValueError, TypeError, ZeroDivisionError):
             # test_arithmetic_explicit_conversions
             return super()._arith_method(other, op)
+
+    def take(
+        self,
+        indices,
+        axis: Axis = 0,
+        allow_fill: bool = True,
+        fill_value=None,
+        **kwargs,
+    ):
+        if kwargs:
+            nv.validate_take((), kwargs)
+        if is_scalar(indices):
+            raise TypeError("Expected indices to be array-like")
+        indices = ensure_platform_int(indices)
+
+        # If allow_fill=True and fill_value=None, just ignore allow_fill,
+        # without raising an exception, as it's done in the base class.
+        if allow_fill and fill_value is not None:
```

Comment:
could we just call _maybe_disallow_fill here?

Reply:
Done, I also added a comment to make it clear that allow_fill isn't used, but we want to raise an exception if it's not valid